CMakeLists.txt

```diff
@@ -393,6 +393,7 @@ if(WITH_OPTIM)
             add_feature_info(ACLE_CRC 1 "Support CRC hash generation using the ACLE instruction set, using \"${ACLEFLAG}\"")
         endif()
         if(WITH_NEON)
+            set(NEONFLAGS "${NEONFLAGS} -DWITH_NEON")
```

Comment:
All ARM-related external defines should start with "ARM_" ... Look at the isolated example I gave in the latest PR by me.

Reply:
Thank you for advice. It's reasonable that the flag starts with ARM. But I don't think it's good to name it ARM_NEON_ADLER32. Since option "--neon" means using the NEON related optimization on ARM platforms, it should not be restricted to adler32_neon. Supposing we have many other optimizations that use NEON, and we have to add many flags like " add_definitions("-DARM_NEON_........") ". So what about  naming it "ARM_USE_NEON" , taking into consideration distinguishing it from the  gcc's parameter "__ARM_NEON__" ?

Reply:
I just backported the define from our internal test version... There is currently only two defines to enable NEON-support in individual functions, in fill_window and adler32, so it's not an issue yet and it makes it really easy to add those to the test suite when it will be implemented.

Reply:
OK ... If so, I change it to ARM_NEON_ADLER32